### ISSUE TITLE
Booting F-Zero AX Rev C or E hangs

### ISSUE THREAD

**Issue by rhester72** [FIX94/Nintendont#210]
Debug log (rev C):

> Nintendont IOS58 v24.32
Built   : Feb 14 2016 15:42:41
Version : 3.387
Game path: /games/F-Zero AX (Rev C)/game.iso
EXIInit Start
Kernel Start
Setup SegaBoot
DIP:DOL EntryPoint::0x80002F8C, GameEntry::0x80005418
Patch:Offset:0x00003200 EOffset:0x000D2FA0 Length:000CFDA0
Patch:Game ID = 47465a52
TRI:SegaBoot (Patched Free Play)
Patch:[SI] applied 18 times
Patch:[AI] applied 17 times
Patch:[DI] applied 43 times
Patch:[PADRead B] applied (0x0001D6B4)
Patch:[ARStartDMA] applied (0x0001ECD0)
Patch:[ARInit] skipped (0x0001EEA4)
Patch:[__ARHandler] skipped (0x0001EF74)
Patch:[__ARChecksize B] applied (0x0001F35C)
Patch:[ARQPostRequest] skipped (0x00020A2C)
Patch:[__DSPHandler] applied (0x0002892C)
Patch:[GXInit] stw r5,-0x7A40(r13) (0x000294CA)
Patch:[__GXSetVAT A] applied (0x0002B628)
Patch:[patch_fwrite_Log] skipped (0x0003A774)
Patch:[EXIImm A] applied (0x0004290C)
Patch:[EXIDMA A] applied (0x00042C08)
Patch:[EXISync B] applied (0x00042CF4)
Patch:[__EXIProbe A] applied (0x00043050)
Patch:[EXISelect A] applied (0x0004338C)
Patch:[EXIDeselect A] applied (0x000434B8)
Patch:[EXIntrruptHandler B] applied (0x000435CC)
Patch:[TCIntrruptHandler B] applied (0x00043694)
Patch:[EXILock A] applied (0x00043B4C)
Patch:[EXIUnlock A] applied (0x00043C98)
Patch:[CompleteTransfer A] applied (0x00044774)
Patch:[SIEnablePollingInterrupt A] applied (0x00044DE4)
Patch:[SIEnablePollingInterrupt A] applied (0x00044DE8)
Patch:[SIInit A] applied (0x00045034)
Patch:[SIInterruptHandler] applied (0x00044A40)
Patch:[SIInterruptHandler] applied (0x00044B6C)
Patch:[__SITransfer A] applied (0x000450E8)
Patch:[__SITransfer A] applied (0x000451D0)
Patch:[SITransfer] applied (0x000456BC)
Patch:[SIGetType A] applied (0x00045AC0)
Patch:[SetInterruptMask] applied (0x00051280)
Patch:[__OSDispatchInterrupt] applied (0x00051668)
Patch:[__OSResetSWInterruptHandler] applied (0x00053158)
Patch:[__DVDInterruptHandler]: 0x00055C44 (0x00055C44)
Patch:[VIConfigure] applied (0x0005B2E0)
Patch:[GCAMSendCommand] applied (0x0005E188)
Patch:[DSP v3] patched (0x00071AC0)
Patch:[PADInit] applied (0x0001D6B0)
Patch:[GXInitTlutObj] not found
Patch:[GXLoadTlut] not found
Patch:[C_MTXPerspective] not found
Patch:[C_MTXLightPerspective] not found
Patch:[OSGetResetButtonState] not found
Patch:[__OSInitAudioSystem] not found
Patch:[PADControlAllMotors] not found
Patch:[PADControlMotor] not found
Patch:[PADIsBarrel] not found
Patch:[__CARDReadStatus] not found
Patch:[__CARDClearStatus] not found
Patch:[__CARDStat] not found
Patch:[ReadROM] not found
Patch:Using a EXI Timing of 0
Jumping to 0x80005418
TRI:Patched boot.id Video Mode
Patch:Offset:0x01300000 EOffset:0x0130FEE0 Length:0000FEE0
Patch:Game ID = 47465a52
Patch:[SI] applied 12 times
Patch:[AI] applied 3 times
Patch:[DI] applied 44 times
Patch:[__DVDInterruptHandler]: 0x01300AE4 (0x01300AE4)
Patch:[SetInterruptMask] applied (0x01303D30)
Patch:[__OSDispatchInterrupt] applied (0x01304118)
Patch:[__OSResetSWInterruptHandler] applied (0x01304990)
Patch:[EXIImm B] applied (0x013057DC)
Patch:[EXIDMA B] applied (0x01305AD8)
Patch:[EXISync C] applied (0x01305BC4)
Patch:[__EXIProbe B] applied (0x01305E14)
Patch:[EXISelect B] applied (0x01305F88)
Patch:[EXIDeselect B] applied (0x013060B4)
Patch:[EXIntrruptHandler C] applied (0x013061CC)
Patch:[TCIntrruptHandler C] applied (0x01306294)
Patch:[EXILock B] applied (0x01306680)
Patch:[EXIUnlock B] applied (0x013067D4)
Patch:[CompleteTransfer A] applied (0x01306B20)
Patch:[SIInit D] applied (0x0130717C)
Patch:[SIInterruptHandler] applied (0x01306DEC)
Patch:[SIInterruptHandler] applied (0x01306F18)
Patch:[__SITransfer A] applied (0x0130722C)
Patch:[__SITransfer A] applied (0x01307314)
Patch:[SITransfer] applied (0x013075A4)
Patch:[SIGetType A] applied (0x013079A8)
Patch:[patch_fwrite_Log] skipped (0x0130B198)
Patch:[SIInit] applied (0x013071C8)
Patch:[__GXSetVAT] not found
Patch:[GXInitTlutObj] not found
Patch:[GXLoadTlut] not found
Patch:[ARStartDMA] not found
Patch:[ARInit] not found
Patch:[__ARChecksize] not found
Patch:[__ARHandler] not found
Patch:[ARQPostRequest] not found
Patch:[C_MTXPerspective] not found
Patch:[C_MTXLightPerspective] not found
Patch:[OSGetResetButtonState] not found
Patch:[__OSInitAudioSystem] not found
Patch:[__DSPHandler] not found
Patch:[GCAMSendCommand] not found
Patch:[SIEnablePollingInterrupt] not found
Patch:[PADRead] not found
Patch:[PADControlAllMotors] not found
Patch:[PADControlMotor] not found
Patch:[PADIsBarrel] not found
Patch:[__CARDReadStatus] not found
Patch:[__CARDClearStatus] not found
Patch:[__CARDStat] not found
Patch:[ReadROM] not found
DIP:DOL EntryPoint::0x80002F8C, GameEntry::0x80003154
Patch:Offset:0x00003100 EOffset:0x00448800 Length:00445700
Patch:Game ID = 47465a52
TRI:F-Zero AX (Rev C)
Patch:[SI] applied 19 times
Patch:[AI] applied 17 times
Patch:[DI] applied 44 times
Patch:[__DVDInterruptHandler]: 0x00184B0C (0x00184B0C)
Patch:[ReadROM] applied (0x00188970)
Patch:[SetInterruptMask] applied (0x00189458)
Patch:[__OSDispatchInterrupt] applied (0x00189840)
Patch:[__OSResetSWInterruptHandler] applied (0x0018B53C)
Patch:[EXIImm A] applied (0x0018DBC4)
Patch:[EXIDMA A] applied (0x0018DEC0)
Patch:[EXISync B] applied (0x0018DFAC)
Patch:[__EXIProbe A] applied (0x0018E2BC)
Patch:[EXISelect A] applied (0x0018E72C)
Patch:[EXIDeselect A] applied (0x0018E858)
Patch:[EXIntrruptHandler B] applied (0x0018E96C)
Patch:[TCIntrruptHandler B] applied (0x0018EA34)
Patch:[EXILock A] applied (0x0018EEEC)
Patch:[EXIUnlock A] applied (0x0018F038)
Patch:[CompleteTransfer A] applied (0x0018FC2C)
Patch:[SIEnablePollingInterrupt A] applied (0x0019029C)
Patch:[SIEnablePollingInterrupt A] applied (0x001902A0)
Patch:[SIInit A] applied (0x001904EC)
Patch:[SIInterruptHandler] applied (0x0018FEF8)
Patch:[SIInterruptHandler] applied (0x00190024)
Patch:[__SITransfer A] applied (0x001905A0)
Patch:[__SITransfer A] applied (0x00190688)
Patch:[SITransfer] applied (0x00190B84)
Patch:[SIGetType A] applied (0x00190F88)
Patch:[VIConfigure] applied (0x001962C4)
Patch:[PADControlAllMotors A] applied (0x001979E4)
Patch:[PADControlMotor B] applied (0x00197A9C)
Patch:[ARStartDMA] applied (0x00198F1C)
Patch:[ARInit] skipped (0x001990E8)
Patch:[__ARHandler] skipped (0x001991AC)
Patch:[__ARChecksize B] applied (0x00199594)
Patch:[ARQPostRequest] skipped (0x0019AC58)
Patch:[__DSPHandler] applied (0x001A35B8)
Patch:[GXInit] stw r5,-0x4858(r13) (0x001A46D2)
Patch:[PI_FIFO_WP] rlwinm r6,r6,0,3,1 (0x001A5A40)
Patch:[PI_FIFO_WP] rlwinm r0,r0,0,3,1 (0x001A5ABC)
Patch:[PI_FIFO_WP] rlwinm r3,r0,0,3,1 (0x001A5CCC)
Patch:[PI_FIFO_WP] rlwinm r6,r0,0,3,1 (0x001A5DC0)
Patch:[__GXSetVAT A] applied (0x001A73BC)
Patch:[PI_FIFO_WP] extrwi r30,r0,1,2 (0x001AC198)
Patch:[GCAMSendCommand] applied (0x001AEAAC)
Patch:[patch_fwrite_Log] skipped (0x001CA63C)
Patch:[__CARDUnlock DBG] applied (0x00209194)
Patch:[DSP v4] patched (0x002FED00)
Patch:[SIInit] applied (0x0019053C)
Patch:[GXInitTlutObj] not found
Patch:[GXLoadTlut] not found
Patch:[OSGetResetButtonState] not found
Patch:[__OSInitAudioSystem] not found
Patch:[PADRead] not found
Patch:[PADIsBarrel] not found
Patch:[__CARDReadStatus] not found
Patch:[__CARDClearStatus] not found
Patch:[__CARDStat] not found
Patch:Using a EXI Timing of 0
Jumping to 0x80003154

I noted that Unk, Rev C and Rev E all attempt to jump into code at address 0x80003154 - since Unk works, is it possible that the jump address should be different for revs C and E?  Just a wild-assed guess, not quite sure where the problem is - if Unk works, it follows the other two ought to as well, since they are very largely the same code base (at least as far as MAME romcmp is concerned).

**Comment by FIX94** [comment]
the differences between the 3 are very small and I'm pretty sure that all patches are for each version:
https://github.com/FIX94/Nintendont/blob/master/kernel/TRI.c
also I did obviously try all versions myself before releasing it and just now tested it again with only memory card emulation enabled and it works just fine. Did you make sure the files you have are good with triforce header patcher?
https://github.com/FIX94/triforce-header-patcher

**Comment by rhester72** [comment]
My apologies, I was unaware of the patcher tool, which worked perfectly (as do the dumps!).  Closing as can't reproduce.  Thank you!

**Comment by FIX94** [comment]
thats quite surprising you didnt know about that considering you had decrypted C and E files which I assume you got from my triforce-iso-extract tool but yea, I never really pushed both tools too much so they are somewhat unknown I guess.

**Comment by rhester72** [comment]
I found the extract tool completely by accident when I was researching the information needed to write one myself, assuming one didn't exist.  LOL  I didn't know about the patcher at all until you mentioned it.  Ironically, the ID6 values I picked for my local conversion efforts precisely matched the ones you picked, so I guess we share brainwaves or something.  LMFAO

Thanks again for the amazing and staggering work you've done on Nintendont!
